### PR TITLE
Adjust output of --help text to mirror command line changes

### PR DESCRIPTION
**Pull request by Michael0x2a**
This commit adjusts the output of running `mypy --help` to mirror the changes made in https://github.com/python/mypy/pull/5333.

Specifically, it rearranges the order of the flags slightly and replaces the summary of the flags at the top with a short description of how to use mypy along with links for learning more.

### CODE REVIEW

**Comment by Michael0x2a** [comment]
For reference, the new output: https://gist.github.com/Michael0x2a/0bdefea47267b59dd7d3810088f88669

Note that it links to a page introduced in https://github.com/python/mypy/pull/5333.

File: mypy/main.py [hunk 1]
```diff
@@ -297,6 +294,37 @@ def infer_python_version_and_executable(options: Options,
         options.python_executable = special_opts.python_executable
 
 
+HEADER = """%(prog)s [-h] [-v] [-V] [more options; see below]
+            [-m MODULE] [-p PACKAGE] [-c PROGRAM_TEXT] [files [files ...]]"""
```
Comment (gvanrossum) [inline mypy/main.py]: Isn't `[[files [files ...]]` redundant? I'd just say `[files ...]`.

**Comment by Michael0x2a** [comment]
@gvanrossum -- I 100% agree with your comments regarding capitalization -- I was about to do it the way you suggested back when I made my first wave of command line improvements, but reluctantly gave up partly because of things like `usage:` and partly because it seemed like it went against argparse's conventions.

Anyways, I'll make the fixes you suggested later tonight.

**Comment by Michael0x2a** [comment]
@gvanrossum -- this is ready for another look. The new help text looks like this: 

https://gist.github.com/Michael0x2a/7220f9a088aac51bbe13f9078d521793

**Pull request merged**